File: C++_Programming/Practise_Programs/PP_021.cpp
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_021
//
//  Description       : This program demonstrates dynamic memory
//                      allocation of an object and accessing its
//                      public data member using the arrow operator.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 15/09/2026
//
//  Time Complexity   : O(1)
//  Space Complexity  : O(n)
//
//////////////////////////////////////////////////////////////////

#include<iostream>
using namespace std;

#pragma pack(1)
class ArrayX
{
    public:
        int *Arr;
        int iSize;

        ArrayX(int X)
        {
            iSize = X;              
            Arr = new int[iSize];   
        }

        ~ArrayX()
        {
            delete []Arr;
        }
};

int main()
{ 
    ArrayX *aobj1 = new ArrayX(5);
    
    cout<<aobj1->iSize<<"\n";

    delete aobj1;
    
    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      No input required.
//
//  Sample Output :
//      5
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_021.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_021
//
//  Description       : This program accepts a number from the user
//                      and checks whether the number is prime or not
//                      using a boolean flag.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 09/07/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

import java.util.Scanner;

class NumberX
{
    //////////////////////////////////////////////////////////////////
    //
    //  Method Name      : CheckPrime()
    //
    //  Description      : It is used to check whether the given
    //                     number is prime or not using a boolean
    //                     flag and checking its divisibility
    //                     from 2 up to half of the given number.
    //
    //  Author           : Shubham Somanath Gadhe
    //  Date             : 09/07/2026
    //
    //////////////////////////////////////////////////////////////////

    public boolean CheckPrime(int iNo)
    {
        int iCnt = 0;
        boolean bFlag = true;

        for(iCnt = 2; iCnt <= (iNo / 2); iCnt++)
        {
            if((iNo % iCnt) == 0)
            {
                bFlag = false;
                break;
            }
        }

        return bFlag;
    }
}

class PP_021
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;
        boolean bRet = false;

        System.out.println("Enter number : ");
        iValue = sobj.nextInt();

        NumberX nobj = new NumberX();

        bRet = nobj.CheckPrime(iValue);

        if(bRet == true)
        {
            System.out.println("It is prime");
        }
        else
        {
            System.out.println("It is not prime");
        }
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input  :
//      Enter number :
//      17
//
//  Sample Output :
//      It is prime
//
//////////////////////////////////////////////////////////////////
